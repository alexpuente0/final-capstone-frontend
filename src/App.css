* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Roboto', sans-serif;
}

.App {
  display: flex;
  flex-direction: row;
  align-items: center;
  background-image: url(./Assets/background.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  min-height: 100vh;
}

.bground {
  background-image: url(./Assets/background.jpg);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  height: 100vh;
  width: 100vw;
}

.hidden {
  display: none;
}

.navigator {
  display: flex;
  flex-direction: column;
  background-color: rgba(0, 128, 0, 0.85);
  text-decoration: none;
  list-style: none;
  height: 430px;
  padding: 100px 15px 15px 15px;
  position: absolute;
  z-index: 1001;
  gap: 30px;
  top: 0;
  left: 0;
  width: 200px;
  border-radius: 0 0 25% 0;
}

.navigator a {
  text-decoration: none;
  color: white;
  font-size: 20px;
  font-weight: 800;
  filter: drop-shadow(-2px 2px 2px rgba(0, 0, 0, 0.9));
}

.ordered {
  display: flex;
  flex-direction: row;
}

.header {
  display: flex;
  padding-top: 60px;
  justify-content: center;
}

.spinner {
  padding-top: 50px;
}

.hamburger {
  height: 45px;
  width: 45px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  position: fixed;
  z-index: 1002;
  cursor: pointer;
  left: 25px;
  top: 25px;
}

.slice {
  height: 40px;
  width: 40px;
  background-image: url(./Assets/icon.jpg);
  background-size: cover;
  border-radius: 50%;
  filter: drop-shadow(-2px 2px 2px #000);
}

.invisible {
  display: none;
}

@media only screen and (max-width: 400px) {
  .navigator {
    position: absolute;
    display: flex;
    flex-direction: column;
    z-index: 1001;
    height: 100vh;
    gap: 20px;
    top: 0;
    left: 0;
    width: 100%;
    align-items: center;
  }

  .spinner {
    padding-top: 50px;
  }

  .header {
    display: flex;
    padding-top: 80px;
    justify-content: center;
  }
}
