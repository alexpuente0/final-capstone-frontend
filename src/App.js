import {
  Route, Routes, useLocation, useNavigate,
} from 'react-router-dom';
import './App.css';
import Header from './components/Header';
import PrivateRoute from './components/PrivateRoute';
import history from './helpers/history';
import HomePage from './pages/HomePage';
import ItemDetailPage from './pages/ItemDetailPage';
import LoginPage from './pages/LoginPage';
import NewItemPage from './pages/NewItemPage';
import ReservationsAddPage from './pages/ReservationsAddPage';
import ReservationsPage from './pages/ReservationsPage';
import DeleteItemPage from './pages/DeleteItemPage';
import SignupPage from './pages/SignupPage';

function App() {
  history.navigate = useNavigate();
  history.location = useLocation();

  return (
    <div className="App">
      <Header />
      <Routes>
        <Route path="/" element={<HomePage />} />
        <Route path="/reservations" element={<PrivateRoute><ReservationsPage /></PrivateRoute>} />
        <Route path="/login" element={<LoginPage />} />
        <Route path="/signup" element={<SignupPage />} />
        <Route path="/items/:id" element={<ItemDetailPage />} />
        <Route path="/new" element={<PrivateRoute><NewItemPage /></PrivateRoute>} />
        <Route path="/reservations/add" element={<PrivateRoute><ReservationsAddPage /></PrivateRoute>} />
        <Route path="/reservations/add/:id" element={<PrivateRoute><ReservationsAddPage /></PrivateRoute>} />
        <Route path="/delete" element={<PrivateRoute><DeleteItemPage /></PrivateRoute>} />
        {/* <Route path="/" element={<ItemsDeletePage />} /> */}
      </Routes>
    </div>
  );
}

export default App;
