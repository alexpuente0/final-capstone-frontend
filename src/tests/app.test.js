import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { HashRouter as Router } from 'react-router-dom';
import App from '../App';
import setupStore from './setupStore';

describe('Test the App.test components', () => {
  it('renders the App correctly', () => {
    const store = setupStore();
    const app = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });

  it('shows the HomePage when clicking the navbar link', () => {
    const store = setupStore(false);
    const { container } = render(
      <Provider store={store}>
        <Router>
          <App url="/" />
        </Router>
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Reservations'));
    fireEvent.click(screen.getByText('Home'));
    expect(
      container.getElementsByClassName('react-multi-carousel-list'),
    ).toHaveLength(1);
  });

  it('shows the ReservationsPage when clicking the navbar link', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Router>
          <App url="/" />
        </Router>
      </Provider>,
    );
    fireEvent.click(screen.getByText('My Reservations'));
    expect(screen.getByRole('heading')).toHaveTextContent('Reservations Page');
  });

  it('shows the Add Reservations when clicking the navbar link', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Router>
          <App url="/" />
        </Router>
      </Provider>,
    );
    fireEvent.click(screen.getByText('Reserve Green'));
    expect(screen.getByRole('heading')).toHaveTextContent('Add Reservation');
  });

  it('shows the New Item when clicking the navbar link', () => {
    const store = setupStore(false);
    const { container } = render(
      <Provider store={store}>
        <Router>
          <App url="/" />
        </Router>
      </Provider>,
    );
    fireEvent.click(screen.getByText('New Green'));
    expect(
      container.getElementsByClassName('MuiFormControl-root'),
    ).toHaveLength(1);
  });

  it('shows the Delete Items when clicking the navbar link', () => {
    const store = setupStore(false);
    render(
      <Provider store={store}>
        <Router>
          <App url="/" />
        </Router>
      </Provider>,
    );
    fireEvent.click(screen.getByText('Delete Green'));
    expect(screen.getByRole('heading')).toHaveTextContent('Delete a Green');
  });

  // it('shows the Item detail page when clicking the item in carousel', async () => {
  //   const store = setupStore(false);
  //   const { container } = render(
  //     <Provider store={store}>
  //       <Router>
  //         <App url="/" />
  //       </Router>
  //     </Provider>,
  //   );
  //   fireEvent.click(screen.getByText('Home'));
  //   // const carousel = container.getElementsByClassName('react-multi-carousel-track').item(0);
  //   const carousel = container.querySelector('.react-multi-carousel-track');
  //   // console.log(carousel);
  //   const list = carousel.querySelector('.react-multi-carousel-item');
  //   console.log('zzzzzzz', list);
  //   // eslint-disable-next-line no-restricted-syntax
  //   for (const c of list) {
  //     // console.log('yyyyy', c);
  //     // const b = c.querySelector('button');
  //     console.log('yyyyy', c.textContent);
  //   }
  //   // fireEvent.click(list[1]);
  //   // console.log('LIST', list[1]);
  //   // list[0].click();
  //   // console.log('CCC', carousel[0]);
  //   // (carousel[0]).click;
  //   // console.log('sssss', await screen.findByRole('button'));
  //   // eslint-disable-next-line no-restricted-syntax
  //   // console.log('yyyyyyyyyyy', carousel.getElementsByClassName('MuiCardMedia-root'));

  //   // console.log('xxxxxxxxxxxxx', screen.getByText('BMW i4'));
  //   // fireEvent.click(screen.getByText('BMW i4'));

  //   expect(screen.getByRole('heading')).toHaveTextContent('BMW i4');
  // });
});
