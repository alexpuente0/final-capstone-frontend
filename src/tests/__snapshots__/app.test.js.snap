// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the App.test components renders the App correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header>
          <nav>
            <ul>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="#/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="#/reservations"
                >
                  My Reservations
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="#/reservations/add"
                >
                  Reserve Green
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="#/new"
                >
                  New Green
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="#/delete"
                >
                  Delete Green
                </a>
              </li>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="#/"
                >
                  Log Out
                </a>
              </li>
            </ul>
          </nav>
        </header>
        <h1>
           
          Hello,
           
          Roberto
          !
           
        </h1>
        <div
          class="react-multi-carousel-list  carousel-container"
          dir="ltr"
        >
          <ul
            class="react-multi-carousel-track "
            style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header>
        <nav>
          <ul>
            <li>
              <a
                aria-current="page"
                class="active"
                href="#/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class=""
                href="#/reservations"
              >
                My Reservations
              </a>
            </li>
            <li>
              <a
                class=""
                href="#/reservations/add"
              >
                Reserve Green
              </a>
            </li>
            <li>
              <a
                class=""
                href="#/new"
              >
                New Green
              </a>
            </li>
            <li>
              <a
                class=""
                href="#/delete"
              >
                Delete Green
              </a>
            </li>
            <li>
              <a
                aria-current="page"
                class="active"
                href="#/"
              >
                Log Out
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <h1>
         
        Hello,
         
        Roberto
        !
         
      </h1>
      <div
        class="react-multi-carousel-list  carousel-container"
        dir="ltr"
      >
        <ul
          class="react-multi-carousel-track "
          style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
